src/legacy/core_plugins/tile_map/common/ems_client.js

```diff
@@ -22,6 +22,7 @@ import _ from 'lodash';
 import { TMSService } from './tms_service';
 import { FileLayer } from './file_layer';
 import fetch from 'node-fetch';
+fetch.Promise = require('bluebird');
```

Comment:
is it an issue that this causes a side-effect for entire Kibana, adding that `Promise` constructor on node-fetch?

Reply:
Is `finally` even needed? You could rework `EMSClientV66` to avoid using `finally` with little modification

```
async _getManifest(manifestUrl) {
    let result;
    try {
      const url = extendUrl(manifestUrl, { query: this._queryParams });
      result = await this._fetchWithTimeout(url);
    } catch (e) {
      if (!e) {
        e = new Error('Unknown error');
      }
      if (!(e instanceof Error)) {
        e = new Error(e.data || `status ${e.statusText || e.status}`);
      }
      throw new Error(`Unable to retrieve manifest from ${manifestUrl}: ${e.message}`);
    }

    return result
      ? await result.json()
      : null;
  }
```

and 
```
_fetchWithTimeout(url) {
    return new Promise(
      (resolve, reject) => {
        const timer = setTimeout(
          () => reject(new Error(`Request to ${url} timed out`)),
          this.EMS_LOAD_TIMEOUT
        );
        fetch(url)
          .then(
            response => {
              clearTimeout(timer)
              resolve(response);
            },
            err => {
              clearTimeout(timer)
              reject(err)
            }
          );
      });
  }
```

Reply:
@nreese Your proposed solution for the removal of the `fetch` finally is the correct solution! The `try..catch..finally` is fine thankfully. Using the fetch `finally` initially seemed like a clean option but it's proven to be a hassle since it's not yet supported by node-fetch and may never be supported by ie/edge? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally#Browser_compatibility